src/internal/cfileinstream.cpp

```diff
@@ -22,7 +22,10 @@ CFileInStream::CFileInStream( const fs::path& filePath ) : CStdInStream( mFileSt
      * Note: we need to do this before and after opening the file (https://stackoverflow.com/a/59161297/3497024). */
     mFileStream.rdbuf()->pubsetbuf( nullptr, 0 );
     openFile( filePath );
+// Unbuffered streams are slow for Visual Studio 2015
+#if defined(_MSC_VER) && _MSC_VER != 1900
```

Comment:
Sorry, I just noticed that also the overall condition is not exactly correct. Bit7z can be built also using other compilers, so the condition should be `!defined(_MSC_VER) || _MSC_VER != 1900`. In this way, either the compiler is not MSVC, or if it is MSVC, it is not MSVC 2015. Basically, the code would work as before except for MSVC 2015.